### DISCUSSION TITLE
00. PyTorch Fundamentals - "Matrix multiplication for tensors above 2 dimensions"

### DISCUSSION THREAD

**Discussion by kkdotpy** [mrdbourke/pytorch-deep-learning#967] [Q&A]
I understand how the matrix multiplication for tensors which has dimensions 1 and 2 work. But as I was playing with the tensors whose dimensions are above 2 (Let's assume the dimension is 3). How do I check if the inner dimensions matches in this case?

For instance: Let's say we are trying to multiply the tensors mentioned below in code

``` python
tensor1 = torch.rand(size=(2,3,2))
tensor1
tensor2 = torch.rand(size=(2,2,3))
tensor2
```

How do I know if I can perform matrix multiplication or not in this case?

**Comment by Asnegha** [comment]
3 dimension tensors is a collection of 2 dimension tensors. So for 2 dimension tensors we check the inner dimensions to be same. Similarly for 3 dimension, first check inner dimension of 2d tensors contained in it satisfy matrix multiplication criteria. Then the outer dimension of 3D tensor should be same in both the tensors which we are gonna multiply.
Ex:
tensor1 = torch.rand(size=(2,3,2))
tensor1
tensor2 = torch.rand(size=(2,2,3))
tensor2
Here tensor1 and tensor2 inner 2d tensor dimension (3,2) and (2,3) satisfy matrix multiplication criteria. Also the outer dimension is 3 in both. So we can do tensor multiplication here.
